resolwe_bio/processes/import_data/metadata.py

```diff
@@ -114,42 +184,65 @@ class Output:
-        target = StringField(label="Target class description")
-        n_metas = IntegerField(label="Number of meta attributes")
 
     def run(self, inputs, outputs):
         """Run the analysis."""
 
-        table_path = inputs.src.import_file(imported_format="extracted")
+        collections = Collection.filter(data__id=self.data.id)
+        if not collections:
+            self.error(
+                "Metadata table was not uploaded to a Collection. "
+                "Matching of metadata entries to Sample objects is not possible."
+            )
 
-        extensions = [".csv", ".tab", ".tsv", ".xlsx", ".xls"]
-        path = Path(table_path)
+        path = Path(inputs.src.import_file(imported_format="extracted"))
 
-        if path.suffix in [e.upper() for e in extensions]:
+        # change the file suffix if it is either upper or mixed case
+        if path.suffix.isupper() or (
+            not path.suffix.islower() and not path.suffix.isupper()
+        ):
             path = change_suffix(path)
             self.info(
                 "File extension of the table was replaced with a lower case version."
             )
 
-        if path.suffix not in extensions:
-            self.error(
-                "Unsupported file name extension. Supported extensions "
-                f"are {', '.join(extensions)}."
-            )
-
         try:
-            data = Orange.data.Table.from_file(str(path))
+            df_data = read_tabular_data(path, error=self.error)
         except Exception as err:
-            outputs.table = str(path)
-            self.error(f"Orange is unable to read the provided data table. {err}")
+            self.error(f"It was not possible to read the provided data table. {err}")
 
-        n_samples = len(data)
+        n_samples = len(df_data)
         if n_samples < 1:
             self.error("The uploaded table contains no samples.")
 
+        sample_columns = {
+            "Sample ID": "id",
+            "Sample slug": "slug",
+            "Sample name": "name",
+        }
+
+        sample_headers = df_data.columns.intersection(sample_columns)
+        if len(sample_headers) != 1:
+            self.error(
+                f"The uploaded metadata table needs to contain "
+                f"exactly one of the following columns: "
+                f"{sorted(sample_columns.keys())}."
+            )
+
+        samples = Entity.filter(collection_id=collections[0].id)
+
+        col_samples = {
+            getattr(sample, sample_columns[sample_headers[0]]) for sample in samples
+        }
+
+        df_samples = df_data[sample_headers[0]].values
+        intersection = col_samples.intersection(df_samples)
+
+        if not intersection:
+            self.warning(
+                "None of the samples listed in the uploaded Sample metadata table "
+                "match the Samples in the working Collection."
+            )
+
         outputs.table = str(path)
         outputs.n_samples = n_samples
```

Comment:
This is not true since there could be duplicates. Better use somthing like:

`outputs.n_samples = len(df_data[sample_headers[0]].unique())`